### PR TITLE
feat: add hsl and hsla functions

### PR DESCRIPTION
**Pull request by bsushmith**
Closes: https://github.com/chasefleming/elem-go/issues/140

### CODE REVIEW

File: styles/utils.go [hunk 1]
```diff
@@ -128,3 +128,30 @@ func Seconds(value float64) string {
 func Milliseconds(value int) string {
 	return strconv.Itoa(value) + "ms"
 }
+
+func HSL(hue, saturation, lightness int) string {
+	var builder strings.Builder
+	builder.WriteString("hsl(")
+	builder.WriteString(strconv.FormatInt(int64(hue), 10))
+	builder.WriteString(",")
+	builder.WriteString(strconv.FormatInt(int64(hue), 10))
```
Comment (chasefleming) [inline styles/utils.go:137]: This should be `saturation`

File: styles/utils.go [hunk 2]
```diff
@@ -128,3 +128,30 @@ func Seconds(value float64) string {
...
+	builder.WriteString("%,")
+	builder.WriteString(strconv.FormatInt(int64(hue), 10))
```
Comment (chasefleming) [inline styles/utils.go:139]: This should be `lightness`

File: styles/utils.go [hunk 3]
```diff
@@ -128,3 +128,30 @@ func Seconds(value float64) string {
...
+	builder.WriteString(")")
+	return builder.String()
+}
+
+func HSLA(hue, saturation, lightness int, alpha float64) string {
+	var builder strings.Builder
+	builder.WriteString("hsla(")
+	builder.WriteString(strconv.FormatInt(int64(hue), 10))
+	builder.WriteString(",")
+	builder.WriteString(strconv.FormatInt(int64(hue), 10))
```
Comment (chasefleming) [inline styles/utils.go:150]: Same

File: styles/utils.go [hunk 4]
```diff
@@ -128,3 +128,30 @@ func Seconds(value float64) string {
...
+	builder.WriteString("%,")
+	builder.WriteString(strconv.FormatInt(int64(hue), 10))
```
Comment (chasefleming) [inline styles/utils.go:152]: Same

**Review by chasefleming — CHANGES_REQUESTED** [review]
Thanks for taking this one! Much appreciated. Left a couple comments. Also, two more items:

- Can you update the tests please so the comments would have be caught?
- Can you use `strconv.Itoa` instead of `strconv.FormatInt` like `RBGA`?

Thanks!

File: styles/utils.go (hunk 1 above)
Comment (bsushmith) [reply styles/utils.go]: My bad. I should have found this through test cases at least. Fixed and updated test cases.

File: styles/utils.go (hunk 2 above)
Comment (bsushmith) [reply styles/utils.go]: updated.

File: styles/utils.go (hunk 3 above)
Comment (bsushmith) [reply styles/utils.go]: updated

File: styles/utils.go (hunk 4 above)
Comment (bsushmith) [reply styles/utils.go]: updated.

**Comment by bsushmith** [comment]
Thanks for the feedback!

> * Can you use `strconv.Itoa` instead of `strconv.FormatInt` like `RBGA`?

I have used `strconv.FormatInt` since that seemed to be [slightly faster](https://dev.to/mavensingh/strconvitoa-vs-strconvformatint-vs-fmtsprintf-which-is-good-to-convert-int-to-string-in-golang-3iol) than `strconv.Itoa`.  What would you suggest?

**Comment by chasefleming** [comment]
@bsushmith Ok, I'm good with that then. Thanks for sharing.

**Review by chasefleming — APPROVED** [review]
Looks good, thanks for grabbing this one!

**Pull request merged**